tests/io/test_RiversObject.py

```diff
@@ -0,0 +1,150 @@
+        assert rivers.ucritical == [[0.3], [0.3, 0.3]]
+        assert rivers.qbankfull == [[150.0], [2500.0, 2500.0]]
+        assert rivers.qstagnant == [[50.0], [0.0, 1500.0]]
+        assert rivers.qmin == [[1.0], [1.0, 1.0]]
+        assert rivers.qfit == [[(10.0, 20.0)], [(800.0, 1280.0), (800.0, 1280.0)]]
+        assert rivers.qlevels == [[(100.0, 200.0, 300.0, 400.0)], [(1000.0, 2000.0, 3000.0, 4000.0), (1000.0, 2000.0, 3000.0, 4000.0)]]
+        assert rivers.dq == [[(5.0, 15.0)], [(1000.0, 1000.0), (1000.0, 1000.0)]]
+        assert rivers.version == '1.0'
+    
+    def test_read_rivers_02(self):
+        """
+        Testing read rivers raising an Exception.
+        """        
+        with pytest.raises(Exception) as cm:
+            RiversObject("tests/files/read_rivers_test_no_version.ini")
+        assert str(cm.value) == 'No version information in the file tests/files/read_rivers_test_no_version.ini!'
+
+    def test_read_rivers_03(self):
+        """
+        Testing read rivers raising an Exception.
+        """        
+        with pytest.raises(Exception) as cm:
+            RiversObject("tests/files/read_rivers_test_wrong_version.ini")
+        assert str(cm.value) == 'Unsupported version number 0.0 in the file tests/files/read_rivers_test_wrong_version.ini!'
+
+class Test_read_rivers2():    
+    def test_read_rivers2_01(self):
+        """
+        Testing version
+        """
+        print("current work directory: ", os.getcwd())
+        rivers = RiversObject("tests/files/read_rivers_test_2_0_version.ini")
+        assert rivers.version == '2.0'
+    
+    def test_read_rivers2_02(self):
+        """
+        Testing read_rivers2, collect_values1, collect_values2 and collect_values4.
+        """
+        print("current work directory: ", os.getcwd())
+        rivers = RiversObject("tests/files/read_riversv2_test.ini")
+        assert rivers.branches == ['Branch1', 'Branch2']
+        assert rivers.allreaches == [['Branch1 R1'], ['Branch2 R1', 'Branch2 R2']]
+        assert rivers.qlocations == ['L1', 'L2']
+        assert rivers.normal_width == [[250.0], [250.0, 100.0]]
+        assert rivers.ucritical == [[0.3], [0.3, 0.3]]
+        assert rivers.qstagnant == [[50.0], [0.0, 1500.0]]
+        assert rivers.qfit == [[(10.0, 20.0)], [(800.0, 1280.0), (800.0, 1280.0)]]
+        
+        assert rivers.version == '2.0'
+        
+        assert rivers.autotime == [[False], [False, False]]
+        assert rivers.cdisch == [[(11.0, 21.0)], [(11.0, 21.0), (11.0, 21.0)]]
+        assert rivers.cform == [[2], [2, 2]]
+        assert rivers.hydro_q == [[()], [(), ()]]
+        assert rivers.hydro_t == [[()], [(), ()]]
+        assert rivers.prop_c == [[()], [(), ()]]
+        assert rivers.prop_q == [[()], [(), ()]]
+        assert rivers.tide == [[True], [True, True]]
+        assert rivers.tide_bc == [[()], [(), ()]]
+        
+    
+    def test_read_rivers2_03(self):
```

Comment:
maybe good to move these tests togetehr in one test and parameterize them? e.g. give expect?

Reply:
Refactoring & improvements should be done in separate issue for each component

Reply:
https://issuetracker.deltares.nl/browse/DFAST-113